modules/auxiliary/scanner/http/dir_scanner.rb

```diff
@@ -118,49 +125,65 @@ def run_host(ip)
       queue << testd.strip + '/'
     end
 
+    dictionary_len = queue.length
+    print_status("Using dictionary file '#{datastore['DICTIONARY']}' (#{dictionary_len} entries)")
+
     while(not queue.empty?)
       t = []
       1.upto(nt) do
         t << framework.threads.spawn("Module(#{self.refname})-#{rhost}", false, queue.shift) do |testf|
           Thread.current.kill if not testf
 
+          # Add the delay based on JITTER and DELAY if needs be
+          add_delay_jitter(delay_value,jitter_value)
+
+          vprint_status("Try... #{wmap_base_url}#{tpath}#{testf} (#{wmap_target_host})")
+
           testfdir = testf
           res = send_request_cgi({
-            'uri'  		=>  tpath+testfdir,
-            'method'   	=> 'GET',
-            'ctype'		=> 'text/html'
-          }, 20)
-
+            'uri'    => tpath+testfdir,
+            'method' => 'GET',
+            'ctype'  => 'text/html'
+          }, timeout)
 
           if(not res or ((res.code.to_i == ecode) or (emesg and res.body.index(emesg))))
             if dm == false
               print_status("NOT Found #{wmap_base_url}#{tpath}#{testfdir} #{res.code} (#{wmap_target_host})")
             end
           else
 
-            report_web_vuln(
-              :host	=> ip,
-              :port	=> rport,
-              :vhost  => vhost,
-              :ssl    => ssl,
-              :path	=> "#{tpath}#{testfdir}",
-              :method => 'GET',
-              :pname  => "",
-              :proof  => "Res code: #{res.code.to_s}",
-              :risk   => 0,
-              :confidence   => 100,
-              :category     => 'directory',
-              :description  => 'Directoy found.',
-              :name   => 'directory'
-            )
+            report_web_vuln({
+              :host        => rhost,
+              :port        => rport,
+              :vhost       => vhost,
+              :ssl         => ssl,
+              :path        => "#{tpath}#{testfdir}",
+              :method      => 'GET',
+              :pname       => "",
+              :proof       => "Res code: #{res.code.to_s}",
+              :risk        => 0,
+              :confidence  => 100,
+              :category    => 'directory',
+              :description => 'Directoy found',
+              :name        => 'directory'
+            })
+
+            report_vuln({
+              :host  => rhost,
+              :port  => rport,
+              :proto => 'tcp',
+              :sname => (ssl ? 'https' : 'http'),
+              :name  => self.name,
+              :info  => "Module used #{self.fullname}",
+            })
```

Comment:
hey @timoles this looks good but I'm wondering why you've added in this `report_vuln` on top of the already existing `report_web_vuln `